Blog App: Add Views

rmiyoyo opened the pull request:
## Description
In this project I added the Views associated with the Models and Controllers I already created for my Blog App.

### Project requirements
- Implemented the design from the sneak peek wireframes. I did not add style as it is not compulsory at the moment.
- Analyzed the wireframes deeply and identified all the necessary elements and how a user would navigate from one page to another.    
- Used methods that I have created in the Blog App - processing data in models.

DammyShittu commented on app/views/users/show.html.erb:
```diff
@@ -1,9 +1,17 @@
+        </div>
+  </div>
+
+  <% @user.posts.each do |post| %>
```
- [ ] The posts here should be rendered using the `three_recent_posts` method in the User model, not all user posts

DammyShittu commented on app/views/posts/index.html.erb:
```diff
@@ -1,9 +1,11 @@
-  </body>
-</html>
+<div>
+  <%= render partial: 'shared/user', locals: {user: @user} %>
+
+  <% @posts.each do |post| %>
+    <%= render partial: 'shared/post', locals: {post: post} %>
+    <%= render partial: 'shared/comment', locals: {post: post} %>
+  <% end %>
+
```
- [ ] You are using a comment's partial, but the comments displayed on this post index page should display only the 5 recent comments using the `five_recent_comments`. The partial displays all the comments which is not right.

DammyShittu commented on app/views/shared/_user.html.erb:
```diff
@@ -0,0 +1,12 @@
+  </div>
+
+  <div class="w-4/6 border-solid border-black border-4">
+    <h1><%= user.name %></h1>
```
- [ ] The user on the user index page should be a link to the user show page. Kindly add this link.

DammyShittu commented on app/views/shared/_post.html.erb:
```diff
@@ -0,0 +1,5 @@
+<div id="post" class="border-solid border-4 border-black md:w-1/2 mx-auto w-full mb-5">
+  <h2 class="font-bold ml-3"><%= post.title %></h2>
```
- [ ] The post on the user show page should be a link to the post show page. Kindly add this.

DammyShittu replied:
- [ ] The post on the post index page should be a link to the post show page. Kindly add this link.

DammyShittu commented on app/views/users/show.html.erb:
```diff
@@ -1,9 +1,17 @@
+    <%= render partial: 'shared/post', locals: {post: post} %>
+  <% end %>
+
+  <button class="mx-auto font-bold w-5/12 md:w-1/10 mb-5 border-solid border-4 border-gray-500">See all posts</button>
```
- [ ] This button does not work. When clicked it should take the user to the post index page.

DammyShittu reviewed (changes requested):
Hi @rmiyoyo,

## Highlights
✔️ Styling feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

rmiyoyo commented:
# Work Done After Review
- Fixed button
- Added link to user
- Fixed to render using five_recent_comments and three_recent_posts
- Fixed all other bugs

danmainah commented on app/views/posts/index.html.erb:
```diff
@@ -1,9 +1,10 @@
+  <%= render partial: 'shared/user', locals: { user: @user } %>
+
+  <% @user.three_recent_posts.each do |post| %>
+    <%= render partial: 'shared/post', locals: { post: post } %>
+    <%= render partial: 'shared/comment', locals: { post: post } %>
```
- [ ] @rmiyoyo Great work implementing the changes requested however as the previous reviewer noted you are displaying all comments instead of only displaying the five recent comments using `five_recent_comments` method, kindly consider implementing the suggested changes.

danmainah commented on app/views/shared/_post.html.erb:
```diff
@@ -0,0 +1,7 @@
+<div id="post" class="border-solid border-4 border-black md:w-1/2 mx-auto w-full mb-5">
+  <%= link_to user_posts_path(post) do %>
+    <h2 class="font-bold ml-3"><%= post.title %></h2>
+  <% end %>
```
- [ ] @rmiyoyo Great work implementing the changes requested, however, you added the wrong link, the link should lead to the post-show page.

danmainah reviewed (changes requested):
Hi @rmiyoyo , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Implemented the change requested to ensure the user show page is only showing the three recent posts and the see all posts button works correctly.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

rmiyoyo commented:
# Work Done After Review
- The app shows five recent comments per post
- The app shows the correct link for user

Olamarx reviewed (approved):
Hi @rmiyoyo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media3.giphy.com/media/3oz8xAFtqoOUUrsh7W/giphy.gif"/>
## Highlights
- Changes fixed ✔️
- Linters passed ✔️
- Gitflow ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
